tests/components/homekit/test_homekit.py

```diff
@@ -701,24 +722,32 @@ async def test_homekit_too_many_accessories(hass, hk_driver):
         entity_filter,
         {},
         DEFAULT_SAFE_MODE,
+        HOMEKIT_MODE_BRIDGE,
         advertise_ip=None,
         entry_id=entry.entry_id,
     )
-    homekit.bridge = Mock()
-    # The bridge itself counts as an accessory
-    homekit.bridge.accessories = range(MAX_DEVICES)
+
+    def _mock_bridge(*_):
+        mock_bridge = HomeBridge(hass, hk_driver, "mock_bridge")
+        # The bridge itself counts as an accessory
+        mock_bridge.accessories = range(MAX_DEVICES)
+        return mock_bridge
+
     homekit.driver = hk_driver
+    homekit.driver.accessory = Accessory(hk_driver, "any")
 
     hass.states.async_set("light.demo", "on")
+    hass.states.async_set("light.demo2", "on")
+    hass.states.async_set("light.demo3", "on")
 
     with patch("pyhap.accessory_driver.AccessoryDriver.start_service"), patch(
         "pyhap.accessory_driver.AccessoryDriver.add_accessory"
-    ), patch("homeassistant.components.homekit._LOGGER.warning") as mock_warn, patch(
-        f"{PATH_HOMEKIT}.show_setup_message"
+    ), patch(f"{PATH_HOMEKIT}.show_setup_message"), patch(
+        f"{PATH_HOMEKIT}.accessories.HomeBridge", _mock_bridge
     ):
         await homekit.async_start()
         await hass.async_block_till_done()
-        assert mock_warn.called is True
+        assert "would exceeded" in caplog.text
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)